File: 0026-remove-duplicates-from-sorted-array/solution.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        stack<int> st;
        st.push(nums[0]);
        vector<int>ans;

        for(int i=1;i<nums.size();i++){
            if(nums[i]!=st.top()){
                st.push(nums[i]);
            }  
        }
        int k=st.size();

        while(!st.empty()){
            ans.push_back(st.top());
            st.pop();
        }
        reverse(ans.begin(),ans.end());

        for(int i=0;i<k;i++){
            nums[i]=ans[i];
        }

        return k;
    }
};

File: 0026-remove-duplicates-from-sorted-array/solution.java
class Solution {
    public int removeDuplicates(int[] nums) {
        
        Stack<Integer> st =new Stack<>();

        

        st.push(nums[nums.length-1]);

        for(int i=nums.length-2;i>=0;i--){
            if(nums[i]!=st.peek()){
                st.push(nums[i]);
            }
        }

        int i=0;

        while(st.size()!=0){
            nums[i++]=st.peek();
            st.pop();
        }

        return i;

    }
}
